docs/index.rst

```diff
@@ -34,7 +34,148 @@ ops.pebble module
 ops.testing module
 ==================
 
+Frameworks for unit testing charms in a simulated Juju environment.
+
+Two frameworks are available:
+
+* State-transition testing, which tests the charm's state transitions in response
+  to events. This is the recommended approach. Install ops with the ``testing``
+  extra to use this framework; for example: ``pip install ops[testing]``
+* Harness, which provides an API similar to the Juju CLI. This is a legacy
+  framework, and has issues, particularly with resetting the charm state between
+  Juju events. This is currently included with a base ``ops`` install.
+
+.. note::
+    Unit testing is only one aspect of a comprehensive testing strategy. For more
+    on testing charms, see `Charm SDK | Testing <https://juju.is/docs/sdk/testing>`_.
+
+State-transition testing
+------------------------
+
+Write tests that declaratively define the Juju state all at once, define the
+Juju context against which to test the charm, and fire a single event on the
+charm to execute its logic. The tests can then assert that the Juju state has
+changed as expected.
```

Comment:
There's a lot of text after this point before the developer can see anything in action. I think it'd be great to show a tiny example right here:

```python
def test_start():
    ctx = Context(MyCharm)
    state = State(...something small here...)
    out = ctx.run(ctx.on.start(), state)
    assert isinstance(out.unit_status, ActiveStatus)
```

Maybe we want *something* really simple in the `State` so readers can see that.